### PR TITLE
Scope ProGuard rules to library

### PR DESCRIPTION
**Pull request by TheNodi**
ProGuard rules are really easy to add from different libraries, but they are hard to exclude from a subset of libraries. I'd suggest scope all rules under the Joda namespace, to avoid collisions with other libraries / main packages.

We are stuck with version 2.10.6 because of the insertion of these rules. When keeping the names of everything that implements Serializable (directly or indirectly), a good portion of the code (especially when using Kotlin / Kotlin Serialization) is not obfuscated, despite it being able to run correctly without the rules.
Scoping the rules to the library only, not only makes sense from a packaging perspective, but also allows for better control over what need Serialization and what doesn't.

I've also found out that we do not need the name of the classes to be kept, Joda works correctly without them, because serialization appears to be used only in live environment where the class is renamed but constant.
Better testing might be needed here, but I can't find a point where it differs.
If we want to keep the class name as well, I'd suggest we scope it under the Joda namespace too.

This leads to both more obfuscation and lower bundle size.
Let me know if you need more details on this.

### CODE REVIEW

**Comment by dlew** [comment]
This is a great change!

@ntsik - As the person who originally contributed this change (in #255), can you think of any reason why we'd need to keep class names?

**Comment by ntsik** [comment]
Hm I can't think of any reason why the names would need to be kept. This change seems fine to me. :+1:

**Comment by dlew** [comment]
Actually, now that I think about it - what happens if you serialize a Joda class in the current version (that kept names) to disk, and then try deserializing it with this version (which does not keep names)? I would think that there might be some problems there, especially with nested serialized Joda classes.

**Comment by ntsik** [comment]
Ah right, that's a really good point. Also, I'd be surprised if ProGuard/R8 generated the same name every time, so even obfuscated names might be different between versions.

**Comment by TheNodi** [comment]
Yes, there are no guarantees ProGuard will generate the same names when changing the source code. Any object that can persist serialized cross-versions must keep its original name.

That being said, I'm not sure this should be a concern for this library's rules for a few reasons.
- You can use the entire library without preserving the names.
- Serialization to disk is probably used only in a small subset of projects, compared to all the user's of Joda.
- When using Java serialization to store persistent information, you probably need more rules and you should be aware of the behavior of ProGuard.
- As said before, unfortunately it's kinda complex to exclude rules included by libraries, hence it would be penalizing everybody who's not using persistent storage.

TL;DR: If someone is using Java Serialization to store information to disk, he should include the [ProGuard example](https://www.guardsquare.com/en/products/proguard/manual/examples#serializable) in its own rules (or a scoped version for the objects he is serializing).

Maybe we add this thread under "Troubleshooting" in the readme, if you think this is a common problem.

**Comment by ntsik** [comment]
It's true that not all users would need these rules, but as someone who has run into ProGuard issues with Joda, it's not always easy to debug. You'd need to be familiar with the internals of Joda, Java Serialization, and ProGuard to know how/why to fix such issues. IMO I'd prefer the principle of least surprise here over saving a few bytes.

**Comment by dlew** [comment]
I agree with ntsik. We should avoid laying a trap that we *know* users could run into.

Minimizing class names is a pretty small part of optimization anyways.

**Comment by TheNodi** [comment]
I amended the requested changes.

**Comment by dlew** [comment]
This has been released in v2.10.7.2. Lemme know if it works out correctly!
